syntax = "proto3";
package mrpc;
import "google/protobuf/descriptor.proto";

extend google.protobuf.MessageOptions {
  optional int32 cmd_id = 50000;
  optional bool no_reply = 50001;
}

enum MessageFlag
{
	REQUEST = 0;
	RESPONSE = 1;
	NO_REPLY = 2;
	MESSAGE_FLAG_END = 3;
}

message Context
{
	int32 cmd_id = 1;
	map<int32, int32> flags = 2;
	int32 error_code = 3;
	string trace_id = 4;
}